contracts/examples/SimpleInterestTermsContract.sol

```diff
@@ -99,20 +112,82 @@ contract SimpleInterestTermsContract is TermsContract {
         view
         returns (uint _expectedRepaymentValue)
     {
-        bytes32 parameters = debtRegistry.getTermsContractParameters(agreementId);
-        uint issuanceBlockTimestamp = debtRegistry.getIssuanceBlockTimestamp(agreementId);
+        SimpleInterestParams memory params = unpackParamsForAgreementID(agreementId);
+
+        if (timestamp <= params.startTimestamp) {
+            /* The query occurs before the contract was even initialized so the
+            expected value of repayments is 0. */
+            return 0;
+        } else if (timestamp >= params.endTimestamp) {
+            /* the query occurs beyond the contract's term, so the expected
+            value of repayment is the full principal plus interest. */
+            return params.principalPlusInterest;
+        } else {
+            uint numUnits = determineNumberOfAmortizationUnitsForTimestamp(timestamp, params);
+            return params.principalPlusInterest.mul(numUnits).div(params.termLengthInAmortizationUnits);
+        }
+    }
+
+    function determineNumberOfAmortizationUnitsForTimestamp(
+        uint timestamp,
+        SimpleInterestParams params
+    )
+        internal
+        returns (uint units)
+    {
+      uint delta = timestamp.sub(params.startTimestamp);
+      if (params.amortizationUnitType == AmortizationUnitType.HOURS) {
```

Comment:
Duplicate logic; this should pull the amortization unit length in seconds either from the `SimpleInterestParams` or the `getAmortizationUnitLengthInSeconds` method.

Reply:
Agreed